frontend-ssr/components/cli_git/git_commands_cc.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+interface Props {
+  defaultValue:Array<RegExp> | Array<string>
+}
+interface State {
+  currentCommandIndex: number,
+  message: string,
+  buttonText: string
+}
+
+export default class MatchCommandClassComponent extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+    this.state = {
+      currentCommandIndex: 0,
+      message: '',
+      buttonText: 'Next',
+    };
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const formEl = event.target;
+    const gitCommand = event.target.gitCommand.value;
+    const { defaultValue } = this.props;
+    const { currentCommandIndex } = this.state;
+
+    let isMatch = false;
+    if (defaultValue[currentCommandIndex] instanceof RegExp) {
+      isMatch = (defaultValue[currentCommandIndex] as RegExp).test(gitCommand);
+    } else {
+      isMatch = defaultValue[currentCommandIndex] === gitCommand.trim().toLowerCase();
+    }
+
+    if (!isMatch) {
+      this.setState({
+        message: 'please try again',
+        buttonText: 'Next',
+      });
+    } else if (currentCommandIndex === defaultValue.length - 1) {
```

Comment:
Shouldn't this be `defaultValue.length - 2`. We want to set the text to `Done` if you're on the last step. Right?

Reply:
It means that I set the text to Done, after the user types git commit -m "Some message". Not after git push?

Reply:
Yes, this is how it should work. After git push you no longer show the button, it's done!